web-ng/src/app/components/user-profile-popup/user-profile-popup.component.ts

```diff
@@ -0,0 +1,63 @@
+/**
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the 'License');
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an 'AS IS' BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { Component, ElementRef, OnInit, Inject } from '@angular/core';
+import { MatDialogRef, MAT_DIALOG_DATA } from '@angular/material/dialog';
+import { MatDialog, MatDialogConfig } from '@angular/material';
+import { AuthService } from './../../services/auth/auth.service';
+
+@Component({
+  selector: 'app-user-profile-popup',
+  templateUrl: './user-profile-popup.component.html',
+  styleUrls: ['./user-profile-popup.component.css'],
+})
+export class UserProfilePopupComponent implements OnInit {
+  private readonly matDialogRef: MatDialogRef<UserProfilePopupComponent>;
+  private readonly triggerElementRef: ElementRef;
+
+  constructor(
+    private dialog: MatDialog,
+    matDialogRef: MatDialogRef<UserProfilePopupComponent>,
+    public auth: AuthService,
+    @Inject(MAT_DIALOG_DATA) data: { trigger: ElementRef }
+  ) {
+    this.matDialogRef = matDialogRef;
+    this.triggerElementRef = data.trigger;
+  }
+
+  ngOnInit() {
+    if (this.triggerElementRef) {
+      const matDialogConfig: MatDialogConfig = new MatDialogConfig();
+      const rect = this.triggerElementRef.nativeElement.getBoundingClientRect();
+      matDialogConfig.position = {
+        left: `${rect.left - 190}px`,
+        top: `${rect.bottom + 10}px`,
+      };
+      matDialogConfig.width = '200px';
+      matDialogConfig.height = '200px';
+      this.matDialogRef.updateSize(
+        matDialogConfig.width,
+        matDialogConfig.height
+      );
+      this.matDialogRef.updatePosition(matDialogConfig.position);
+    }
+  }
+
+  close() {
```

Comment:
Naming nit: `close()` => `onSignOut()`? Otherwise it seems this _only_ closes the popup. If we call if "signout" the fact that the popup closes can be inferred. (In actuality we'll eventually redirect to another page anyway).